Sources/MessagingPushAPN/RichPush/MessagingPush+NotificationServiceExtension.swift

```diff
@@ -40,6 +42,18 @@ public extension MessagingPush {
             return false
         }
 
+        // If cioHandleBadge exists and value is true
+        if let cioHandleBadge = pushContent.cioHandleBadge, cioHandleBadge  {
+            
+            let badgeCount = store.badgeCount ?? 0
+            // Update badge count in payload
+            if let newRequest = request.content.mutableCopy() as? UNMutableNotificationContent {
```

Comment:
Can you move this logic of creating `UNMutableNotificationContent` and modifying it into `PushContent` class? That class exists for that reason already.